### ISSUE TITLE
Re-evaluate usage of environment variables for static values in docker-compose.yml

### ISSUE THREAD

**Issue by rfay** [drud/ddev#372]
## What happened (or feature request):

We use environment variables instead of templating in docker-compose.yml, and it seems this adds unnecessary complexity in some cases. 

IMO we should use templating (actually replacing into the docker-compose.yml) for all values that are static. For example there's no reason I understand that these should be driven by environment variables:

```
    container_name: ddev-${DDEV_SITENAME}-db
      - "${DDEV_IMPORTDIR}:/db"
      - "${DDEV_DATADIR}:/var/lib/mysql"
      com.ddev.approot: $DDEV_APPROOT
      com.ddev.app-url: $DDEV_URL
```

They all seem to be static values, known at `ddev start` time and unchanging over the life of the universe.

We do have a couple of actual dynamic variables that are well suited for environment variables and that might be difficult to handle otherwise:
```
      - DDEV_UID=$DDEV_UID
      - DDEV_GID=$DDEV_GID
```

Basically, I propose that all static values be added into the docker-compose via templating, and runtime values be left as environment variables.

## Related source links or issues:

**Comment by tannerjfco** [comment]
The biggest reason I see to use env vars for these is to keep the docker-compose.yaml portable, minimally between developers on the same project. DDEV_IMPORTDIR is /Users/tannerjfco/.ddev/sitename/import for me, but likely not for others. I find it kind of nice that our compose file can work for pretty much any ddev site, and it keeps from having to update the compose template if values in config.yaml change.

The biggest downside I see is that `docker-compose` on its own ends up being pretty much useless for a ddev site.

**Comment by rfay** [comment]
I know we have a working idea that .ddev should be portable, and you're right if it is then this is a super-good reason. I still question that idea. I think it's just a mess of trouble for everybody. I'd think a "ddev config" would be good for everybody's ddev soul.

**Comment by rickmanelius** [comment]
Adding hibernate for now because I would like to see user feedback and how people are ultimately using it and whether this is an issue that comes up.

**Comment by rfay** [comment]
Duplicate of #250 - We do need to move to explicit docker-compose.yaml now instead of peppering things with environment variables. The key thing is we've moved on to writing docker-compose.yaml explicitly.
